src/processor/reconnection-graph.service.ts

```diff
@@ -431,12 +429,47 @@ export class ReconnectionGraphService {
   }
 
   async processChainEvents(promises: Promise<ParsedEventResult>[], dsnpUserId: MessageSourceId): Promise<void> {
-    // loop over promises and wait for all to resolve
-    for (const promise of promises) {
-      const [event, eventMap] = await promise;
-      if (!event) {
-        throw new Error(`Error submitting extrinsic`);
+    // iterate over promises and wait for all to resolve
+    try {
+      await Promise.all(promises);
+      this.logger.debug(`Successfully processed chain events for ${dsnpUserId.toString()}`);
+
+      // iterate over promises and check for `BatchCompleted` event
+      for (const promise of promises) {
+        const [event, eventMap] = await promise;
```

Comment:
We're awaiting these promises twice, aside from being unnecessary, what happens if you await a promise that has already resolved or rejected? Does it get run again?

Reply:
yep simplified it